trainer/diffusers_trainer.py

```diff
@@ -829,109 +831,113 @@ def save_checkpoint(global_step):
-                                tokenizer=tokenizer,
-                                scheduler=scheduler,
-                                safety_checker=None, # disable safety checker to save memory
-                                feature_extractor=CLIPFeatureExtractor.from_pretrained("openai/clip-vit-base-patch32"),
-                            ).to(device)
-                            # inference
-                            if args.enablewandb:
-                                images = []
-                            else:
-                                saveInferencePath = args.output_path + "/inference"
-                                os.makedirs(saveInferencePath, exist_ok=True)
-                            with torch.no_grad():
-                                with torch.autocast('cuda', enabled=args.fp16):
-                                    for _ in range(args.image_log_amount):
-                                        if args.enablewandb:
-                                            images.append(
-                                                wandb.Image(pipeline(
-                                                    prompt, num_inference_steps=args.image_log_inference_steps
-                                                ).images[0],
-                                                caption=prompt)
-                                            )
-                                        else:
-                                            from datetime import datetime
-                                            images = pipeline(prompt, num_inference_steps=args.image_log_inference_steps).images[0]
-                                            filenameImg = str(time.time_ns()) + ".png"
-                                            filenameTxt = str(time.time_ns()) + ".txt"
-                                            images.save(saveInferencePath + "/" + filenameImg)
-                                            with open(saveInferencePath + "/" + filenameTxt, 'a') as f:
-                                                f.write('Used prompt: ' + prompt + '\n')
-                                                f.write('Generated Image Filename: ' + filenameImg + '\n')
-                                                f.write('Generated at: ' + str(global_step) + ' steps' + '\n')
-                                                f.write('Generated at: ' + str(datetime.now().strftime('%Y-%m-%d %H:%M:%S'))+ '\n')
-
-                            # log images under single caption
-                            if args.enablewandb:
-                                run.log({'images': images}, step=global_step)
-
-                            # cleanup so we don't run out of memory
-                            del pipeline
-                            gc.collect()
-                        torch.distributed.barrier()
+                
+                if ((idx + 1) % args.gradient_accumulation_steps == 0) or (idx + 1 == len(train_dataloader)):
```

Comment:
if you invert this if check and instead do a continue, you can keep the indents from creeping up even more.

Reply:
```py
if not ((idx + 1) % args.gradient_accumulation_steps == 0) and not (idx + 1 == len(train_dataloader)):
    continue
```